popde/adaptive_kde.py

```diff
@@ -224,6 +224,143 @@ def optimize_parameters(self, method='loo_cv', fom_plot_name=None):
         return best_params, best_score
 
 
+class KDERescaleOptimization(AdaptiveBwKDE):
```

Comment:
hmm, maybe that won't work with how the class is currently, since it is set up to calculate the per-dimension bandwidths with a simple plug-in function rather than take them as input and optimize. OK, forget about this

Reply:
Maybe this idea work. 

Here I am attaching two plots, One where I use this current version of optimizing class on a two d data:  the resulting optimized values of  re-scaling factors are used in KDE and first plot is showing that.
Second plot is just using  _![Figure_1](https://github.com/user-attachments/assets/2d721656-f8d8-4528-bc04-6b4e33de5cc1)_  that compute rescaling factors and fit kde using that.
![Figure_2](https://github.com/user-attachments/assets/21032fbd-d27c-4d8c-8564-7f0de98f0722)

Reply:
Sorry, I am not following what you are proposing here.  Again it seems like you have a test data set which does not require different rescaling factors between different dimensions. 
My recommendation at the moment is _not_ to use MultiDimRescalingBwKDEPy and also, as stated in the issue, to do optimization for the rescaling factors.